doc/mgmt/SONiC_OC_MGMT_Interface_HLD.md

```diff
@@ -0,0 +1,259 @@
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/09/2019  |   Ravi Vasanthm     | Initial version                   |
+
+# About this Manual
+This document provides general information about openconfig support for Management interface handling in SONIC.
+
+# Scope
+This document describes the high level design of openconfig support for Management interface handling feature. Call out any related design that is not covered by this document
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| MGMT Intf                     | Management Interface                     |
+
+# 1 Feature Overview
+Currently SONIC does not support managing/configuring Management interface parameters via CLI, REST and GNMI. As part of this feature will be providing config set/get and status get support for Management interface via openconfig-interfaces.yang.
+
+## 1.1 Requirements
+
+![](http://10.59.132.240:9009/projects/csg_sonic/documentation/graphics/templates/test1.png)
+
+### 1.1.1 Functional Requirements
+
+1. Provide CLI, REST and GNMI support for configuring and displaying management interface attributes.
+2. Enhance existing implementation of interfaces yang to include management interface handling.
+3. Enhance existing top level show commands for interfaces to include management interface details too.
+
+
+### 1.1.2 Configuration and Management Requirements
+1. Provide CLI/GNMI/REST support for configuring Management interface attributes.
+2. Provide CLI/GNMI/REST support for show Management  interface attributes/parameters.
+
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+Will be enhancing the management framework management backed code and transformer methods to add support for Management interface Handling.
+
+
+### 1.2.2 Container
+All code changes will be done in management-framework container.
+
+### 1.2.3 SAI Overview
+ N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Manage/configure management interface via GNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide GNMI and REST support for get/set of Management interface attributes and CLI config and show commands to manage Management interface.
+
+# 3 Design
+## 3.1 Overview
+1. Transformer common app owns the openconfig-interface.yang models (which  means no separate app module required for interfaces yang objects handling) Will be deleting the existing interface app module.
```

Comment:
Need a period after the `)`.

Reply:
If you're deleting the existing interface app module, then this HLD covers physical interfaces too, not just the management interface, right? If that's the case, then you should revise the scope of this document to include physical interfaces.

Reply:
Addressed